src/main/java/io/deephaven/csv/reading/headers/DelimitedHeaderFinder.java

```diff
@@ -0,0 +1,95 @@
+package io.deephaven.csv.reading.headers;
+
+import io.deephaven.csv.CsvSpecs;
+import io.deephaven.csv.containers.ByteSlice;
+import io.deephaven.csv.reading.cells.CellGrabber;
+import io.deephaven.csv.util.CsvReaderException;
+import io.deephaven.csv.util.MutableBoolean;
+import io.deephaven.csv.util.MutableObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class DelimitedHeaderFinder {
+    /**
+     * Determine which headers to use. The result comes from either the first row of the file or the user-specified
+     * overrides.
+     */
+    public static String[] determineHeadersToUse(final CsvSpecs specs,
```

Comment:
This doesn't need to be public? Can be placed as package private class at same level as CsvReader?

Reply:
I moved a few files into new subpackages `deephaven.csv.reading.cells` and `deephaven.csv.reading.headers`
because I thought `deephaven.csv.reading` was getting a little crowded.

But, I realize Java is a little bit annoying by the way every directory is a subpackage, and one of the evils we have to live with is a lot of .java files in the same directory.

And it's not a lot of files. As of the second PR, there will be three files in `deephaven.csv.reading.cells` and two files in `deephaven.csv.reading.headers`

If you think it is more Java-canonical to move those files back up into the parent package. I can do that. Let me know.

Reply:
I agree, somewhat annoying the way packaging / visibility works in Java 8. Ultimately, my general preference is to reduce the "public API" to ensure we are free to change implementation details without breaking users. That said, deephaven-csv doesn't really follow this convention (lots of stuff that is "public API" that doesn't technically need to be), so I'm happy keeping this as-is. 

If we bump to 9+, we can adopt modules which allows libraries to only export specific packages.